src/D2L.CodeStyle.Analyzers/ApiUsage/ContentPhysicalPaths/PhysicalPathPropertyAnalysis.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System.Collections.Immutable;
+using System.Linq;
+using D2L.CodeStyle.Analyzers.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.ApiUsage.ContentPhysicalPaths {
+
+	internal sealed class PhysicalPathPropertyAnalysis {
+
+		internal static readonly DiagnosticDescriptor DiagnosticDescriptor = Diagnostics.ContentPhysicalPathUsages;
+
+		private readonly string m_dangerousTypeName;
+		private readonly string m_dangerousPropertyName;
+		private readonly IImmutableSet<string> m_whitelistedTypes;
+
+		public PhysicalPathPropertyAnalysis(
+				string dangerousTypeName,
+				string dangerousPropertyName,
+				IImmutableSet<string> whitelistedTypes
+			) {
+
+			m_dangerousTypeName = dangerousTypeName;
+			m_dangerousPropertyName = dangerousPropertyName;
+			m_whitelistedTypes = whitelistedTypes;
+		}
+
+		public void Initialize( AnalysisContext context ) {
+
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction( Register );
+		}
+
+		private void Register( CompilationStartAnalysisContext context ) {
+
+			IImmutableSet<ISymbol> dangerousProperties = GetDangerousPropertySymbols( context.Compilation );
+
+			if( dangerousProperties.Count == 0 ) {
+				return;
+			}
+
+			context.RegisterSyntaxNodeAction(
+				ctxt => {
+					if( ctxt.Node is MemberAccessExpressionSyntax propertyAccess ) {

```

Comment:
So this will only get called when `ctxt.Node is SimpleMemberAccessExpression` is true... isn't `MemberAccessExpressionSyntax` more general? So I think this `if` is unnecessary because the case should always succeed.

Reply:
This was the only non-nit comment

Reply:
Makes sense. I removed the if-statement. 
To my defense, I copied that code from the `DangerousMemberUsagesAnalyzer` ... so maybe it is not needed there either.

Reply:
Haha probably